File: budget-buddy-front/src/app/entity/MonthlyBudgetOverview.ts
import { BudgetPerCategory } from "./BudgetPerCategory";
import { GraphData } from "./GraphData";
import { ProjectData } from "./ProjectData";

export interface MonthlyBudgetOverview {
  month: string;
  budgetsPerCategory: BudgetPerCategory[];
  graphData: GraphData;
  projectsData: ProjectData[]
}

File: budget-buddy-back/src/main/java/be/yorian/budgetbuddy/dto/MonthlyBudgetOverview.java
package be.yorian.budgetbuddy.dto;

import be.yorian.budgetbuddy.dto.category.BudgetPerCategory;

import java.util.List;

public record MonthlyBudgetOverview(
        String month,
        List<BudgetPerCategory> budgetsPerCategory,
        GraphData graphData,
        List<ProjectData> projectsData) {
}
